src/ui/map.js

```diff
@@ -2162,7 +2162,7 @@ class Map extends Camera {
      *
      * @param {Object} feature Feature identifier. Feature objects returned from
      * {@link Map#queryRenderedFeatures} or event handlers can be used as feature identifiers.
-     * @param {string | number} feature.id Unique id of the feature.
+     * @param {string | number} feature.id Unique id of the feature. Supports strings values only when the [`promoteId`](https://docs.mapbox.com/mapbox-gl-js/style-spec/sources/#vector-promoteId) option has been applied to the source.
```

Comment:
A few suggestions in here:

- Lead with `number` when listing types since that will always work
- Clarifies that `number` must be an integer (if that's still true?)
- Clarifies that a `string` will also work if "the string can be cast to an integer" (if that's still true?)

```suggestion
     * @param {number | string} feature.id Unique id of the feature. Can be an integer or a string, but supports string values only when the [`promoteId`](https://docs.mapbox.com/mapbox-gl-js/style-spec/sources/#vector-promoteId) option has been applied to the source or the string can be cast to an integer.
```